cql3/selection/selection.cc

```diff
@@ -503,10 +503,12 @@ selection::collect_metadata(const schema& schema, const std::vector<prepared_sel
 }
 
 result_set_builder::result_set_builder(const selection& s, gc_clock::time_point now,
-                                       std::vector<size_t> group_by_cell_indices)
+                                       std::vector<size_t> group_by_cell_indices,
+                                       std::optional<uint64_t> limit)
```

Comment:
You could send numeric_limits::max here and avoid the optional and some complexity.

Reply:
Wait I already made this comment. I'm so repeatable.

Reply:
Checking for `numeric_limits::max` obscures the logic, whereas `std::optional` clearly states what to expect, do we really need this micro-optimization? "Premature optimization..."